src/edgeworth.jl

```diff
@@ -77,12 +79,14 @@ end
 struct EdgeworthSum{D<:UnivariateDistribution} <: EdgeworthAbstract
     dist::D
     n::Float64
-    function EdgeworthSum{D}(d::T, n::Real) where {D<:UnivariateDistribution,T<:UnivariateDistribution}
-        @check_args(EdgeworthSum, n > zero(n))
+    function EdgeworthSum{D}(d::UnivariateDistribution, n::Real; check_args::Bool=true) where {D<:UnivariateDistribution}
```

Comment:
Maybe just
```suggestion
    function EdgeworthSum{D}(d::UnivariateDistribution, n::Real; check_args::Bool=true)
```
?

Reply:
We still need some where clause though but AFAIK the type restriction in the where clause could be dropped. I don't think it matters much, I guess the approach in the PR (and on master) ensures that the constructor can't be called if `D` is not a subtype of `UnivariateDistribution` whereas without the restriction in the where clause the call to `new` would error.